cfme/tests/cloud/test_providers.py

```diff
@@ -1486,3 +1497,47 @@ def test_add_second_provider(setup_provider, provider, request):
     second_provider.refresh_provider_relationships()
     second_provider.validate_stats(ui=True)
     assert provider.exists and second_provider.exists
+
+
+@test_requirements.cloud
+@pytest.mark.meta(automates=[1632750], blockers=[BZ(1632750,
+                                                 unblock=lambda child_provider: "object_managers"
+                                                 in child_provider)])
+@pytest.mark.uncollectif(lambda child_provider, provider:
+                        (provider.one_of(EC2Provider) and "object_managers" in child_provider) or
+                        (provider.one_of(AzureProvider) and ("object_managers" or "block_managers"
+                                                             in child_provider)),
+                        reason="Storage is not supported by AzureProvider "
+                               "and Object Storage is not supported by EC2Provider")
+@pytest.mark.provider([AzureProvider, EC2Provider, OpenStackProvider], scope="function")
+def test_cloud_provider_dashboard_after_child_provider_remove(
+        appliance, provider, request, setup_provider_funcscope, child_provider):
+    """
+        Bugzilla: 1632750
+
+        Polarion:
+        assignee: mmojzis
+        casecomponent: Cloud
+        initialEstimate: 1/6h
+        caseimportance: high
+        casecomponent: Cloud
+        testSteps:
+            1. Have a cloud provider added
+            2. Delete one of its child managers
+            3. Go to cloud provider Dashboard
+        expectedResults:
+            1.
+            2.
+            3. Dashboard should load without any issues
+    """
+    child_provider.delete()
+
+    @request.addfinalizer
+    def _wait_for_delete_provider():
+        provider.delete()
+        provider.wait_for_delete()
```

Comment:
This shouldn't be necessary, the current provider will be removed before the next one is setup.

Reply:
Well, it didn't work for me for a while without waiting for delete but now it does so I removed it.

Reply:
Okay, it happened again. When I tried to run test again it told me it has no managers as they were deleted in previous run. For some reason sometimes provider is not deleted so I will rather keep it there to make it flawless.

Reply:
Please add in-lie comments concerning the behavior you saw and how this fixed it, for future test owners sanity :)

Reply:
Added